pkg/plugins/manager/manager.go

```diff
@@ -683,3 +766,90 @@ func (pm *PluginManager) GetDataPlugin(id string) plugins.DataPlugin {
 func (pm *PluginManager) StaticRoutes() []*plugins.PluginStaticRoute {
 	return pm.staticRoutes
 }
+
+func (pm *PluginManager) Install(ctx context.Context, pluginID, version string) error {
+	plugin := pm.GetPlugin(pluginID)
+	if plugin != nil {
+		if plugin.Info.Version == version {
+			return plugins.DuplicatePluginError{
+				PluginID:          pluginID,
+				ExistingPluginDir: plugin.PluginDir,
+			}
+		}
+
+		// remove existing installation of plugin
+		err := pm.Uninstall(context.Background(), plugin.Id)
```

Comment:
As far as I know there are several plugin directories checked on start (may be wrong), is it ok if plugin removed from one dir but have been downloaded to another?